src/basic/time-util.c

```diff
@@ -910,14 +908,36 @@ int parse_timestamp(const char *t, usec_t *usec) {
         }
 
         if (pid == 0) {
-                if (setenv("TZ", timezone, 1) != 0) {
+                bool with_tz = true;
+
+                if (setenv("TZ", t_timezone, 1) != 0) {
                         shared->return_value = negative_errno();
                         _exit(EXIT_FAILURE);
                 }
 
                 tzset();
 
-                shared->return_value = parse_timestamp_impl(t, &shared->usec, true);
+                //If there is a timezone that mates the tzname fields, leave the parsing to the implementation
+                //Otherwise just cut if of
+                if (t_timezone) {
```

Comment:
Why this whole check is necessary? We could only reach this line if we have a valid IANA timezone (see the `if (last_space != NULL && timezone_is_valid(last_space + 1))` line)

Reply:
I removed the t_timezone check. The rest is still needed, since a valid timezone can differ from tzname[x].

Reply:
@marcelhollerbach if there is a valid IANA timezone name specified in the time string - then it should be used exclusively. `tzname` is not useful at that point anymore.

Reply:
@zerkms see my reply in https://github.com/systemd/systemd/pull/6879#discussion_r140231002